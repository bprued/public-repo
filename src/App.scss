.app-container {
  padding: 48px;
  min-height: 280px;
  background: rgb(255, 255, 255);
  h1 {
    margin-bottom: 25;
  }
}
