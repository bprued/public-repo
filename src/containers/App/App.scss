.app-container {
  padding: 48px;
  min-height: 280px;
  h1 {
    margin-bottom: 25;
  }
}
