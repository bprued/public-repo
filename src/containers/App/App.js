import React, { useState } from "react";
import { useQuery } from "react-query";
import axios from "axios";
import './App.scss';

import Table from "../../componnents/Table";
import Pagination from "../../componnents/Pagination";

const ROW_PER_PAGE = 10

const App = () => {
  const [currentPage, setCurrentPage] = useState(1)
  const [totalPage, setTotalPage] = useState(1)
  const { isLoading, error, data, isFetching } = useQuery("repoData", () =>
    axios.get(
      "https://api.github.com/repositories"
    ).then((res) => {
      setTotalPage(res.data.length/ROW_PER_PAGE)
      return res.data
    })
  );

  if (error) return "An error has occurred: " + error.message;

  const getDataPerPage = (page) => {
    let dataPage = []
    page = page -1
    if(page >= 0 && page < totalPage) {
      dataPage = data.slice(page*ROW_PER_PAGE, ((page+1)*ROW_PER_PAGE))
    }
    return dataPage
  }

  return (
    <div className="app-container">
      {isLoading? <h1>Loading...</h1> : 
      <>
        <h1>Github Public Repositories</h1>
        {isFetching ? "Updating..." : 
          <>
            <Table data={getDataPerPage(currentPage)} />
            <Pagination currentPage={currentPage} totalPage={totalPage} setCurrentPage={setCurrentPage} />
          </>
        }
      </>
      }
    </div>
  );
}

export default App;
