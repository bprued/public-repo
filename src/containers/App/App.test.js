import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from "react-query";
import App from './App';

jest.mock('react-query', () => {
  const original = jest.requireActual("react-query")
  return {
    ...original,
    useQuery: () => ({
      data: [
        {
          id: 1,
          name: 'test1',
          owner: {
            login: 'user1'
          }
        },
      ]
    }),
  }
})

const queryClient = new QueryClient();

const AppComponent = (
  <QueryClientProvider client={queryClient}>
    <App />
  </QueryClientProvider>
)

test('Should match snapshot', () => {
  const { asFragment } = render(AppComponent)
  const titleText = screen.getByText(/Github Public Repositories/i);
  expect(titleText).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot()
})
