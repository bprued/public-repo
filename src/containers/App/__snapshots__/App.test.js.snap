// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-container"
  >
    <h1>
      Github Public Repositories
    </h1>
    <table>
      <thead>
        <tr>
          <th
            style="width: 20%;"
          >
            ID
          </th>
          <th
            style="width: 40%;"
          >
            Name
          </th>
          <th>
            Owner
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            test1
          </td>
          <td>
            user1
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="pagination"
    >
      <button
        disabled=""
      >
        &lt;
      </button>
      <button
        class="active"
      >
        1
      </button>
      <button
        disabled=""
      >
        &gt;
      </button>
    </div>
  </div>
</DocumentFragment>
`;
